sklearn/linear_model/coordinate_descent.py

```diff
@@ -1796,7 +1796,7 @@ def fit(self, X, y):
         X, y, X_offset, y_offset, X_scale = _preprocess_data(
             X, y, self.fit_intercept, self.normalize, copy=False)
 
-        if not self.warm_start or self.coef_ is None:
+        if not self.warm_start or not hasattr(self, "coef_"):
```

Comment:
@jnothman shall I replace `not hasattr(self, "coef_")` by a getattr and we're good?

Reply:
I'm okay with it either way, actually. I just never know how flexible we should make these kinds of changes... I mean if someone's gone and changed their code now to set `coef_ = None` so that it works, using getattr would make sure it continues to work in the next release... but it's also hacky.